File: Dev/C++/Mascaret/src/VEHA/Behavior/BasicActions/SendSignalAction.cpp
#include "VEHA/Behavior/BasicActions/SendSignalAction.h"
#include "VEHA/Behavior/BasicActions/SendSignalBehaviorExecution.h"
#include "VEHA/Behavior/Common/Signal.h"

namespace VEHA
{

SendSignalAction::SendSignalAction()
{
	_kind = "SendSignal";
}

SendSignalAction::~SendSignalAction()
{
}

shared_ptr<BehaviorExecution> SendSignalAction::createBehaviorExecution(shared_ptr<InstanceSpecification> host, const Parameters& p, bool sync)
{
	shared_ptr<SendSignalBehaviorExecution> behavior(new SendSignalBehaviorExecution(shared_dynamic_cast<SendSignalAction>(shared_from_this()),host,p));
	return behavior;
}

bool SendSignalAction::equalsTo(shared_ptr<Behavior> behavior)
{
	shared_ptr<SendSignalAction> sigAction = shared_dynamic_cast<SendSignalAction>(behavior);
	if (sigAction)
	{
		shared_ptr<Signal> signal = sigAction->getSignalClass();
		if (signal) 
		{
			cerr << " &&&&&& " << signal->getName() << " == " << getSignalClass()->getName() << endl;
			if (signal->getName() == getSignalClass()->getName())
			{
				
				string targetName = sigAction->getTarget()->getClassifier()->getName(); 
			}
		}
	}
	else
		return false;
}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Behavior/BasicActions/SendSignalAction.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class SendSignalAction : Action, IEquatable<SendSignalAction>
    {


        private SendSignalTarget target;
        public SendSignalTarget Target
        {
            get { return target; }
            set { target = value; }
        }


        private InstanceSpecification signal;
        public InstanceSpecification Signal
        {
            get { return signal; }
            set { signal = value; }
        }


        private Signal signalClass;
        public Signal SignalClass
        {
            get { return signalClass; }
            set { signalClass = value; }
        }

        public SendSignalAction()
        {
            Kind = "SendSignal";
            target = null;
        }

        public bool Equals(SendSignalAction other)
        {
            if (other == this)
                return true;
            return false;
        }

        //default parameter sync = false
        public override BehaviorExecution createBehaviorExecution(InstanceSpecification host, Dictionary<string, ValueSpecification> p, bool sync)
        {
            SendSignalBehaviorExecution be = new SendSignalBehaviorExecution(this, host, p);
            return be;
        }

    }
}
